mlflow/pyfunc/__init__.py

```diff
@@ -1024,6 +1030,9 @@ def spark_udf(spark, model_uri, result_type="double", env_manager=_EnvManager.VI
     from pyspark.sql.types import DoubleType, IntegerType, FloatType, LongType, StringType
     from mlflow.models.cli import _get_flavor_backend
 
+    # Used in test to force install lcoal version of mlflow when starting a model server
+    mlflow_home = os.environ.get("MLFLOW_HOME")
```

Comment:
The impact of this appears to extend beyond tests: https://github.com/mlflow/mlflow/pull/6575/files#r959175274.

```suggestion
    # Used in test to force install local version of mlflow when starting a model server
    mlflow_home = os.environ.get("MLFLOW_HOME")
```

Reply:
commented above. hopefully it explains it. we can also rename MLFLOW_HOME to something like `MLFLOW_DEV_PROJECT_HOME` in a followup